samples/azure-quantum/qae-numerical-integration/QuantumAmplitudeEstimation.ipynb

```diff
@@ -0,0 +1,1148 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot(ys, k_vals)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this plot we see that all of our results look fairly close to the expected value graphed in green.\n",
+    "\n",
+    "Therefore, we'll graph the errors to get see which $k$ performs best."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def epsilon_squared(x : float, expected : float) -> float:\n",
+    "    return (x - expected) ** 2\n",
+    "\n",
+    "def compute_err(ys : list, vals : list) -> list:\n",
+    "    assert len(ys) == len(vals)\n",
+    "    err = []\n",
+    "    f_y = list(map(f, ys))\n",
+    "    int_f_y = list(map(int_f, ys))\n",
+    "    #int_f_y = list(map(lambda x : sum(f_y[1:x+1]) / n, range(len(ys))))\n",
+    "    for i in range(len(ys)):\n",
+    "        y = ys[i]\n",
+    "        g_y = int_f_y[i]\n",
+    "        g_hat_y = vals[i]\n",
+    "        err.append(epsilon_squared(g_hat_y, g_y))\n",
```

Comment:
This would be much more Python-ic as a list comprehension over `zip(y, g_y, g_hat_y)`. In particular, the index `i` is never used, and thus should be omitted to focus on the actual transformation that you're implementing:

```suggestion
    "    err = [epsilon_squared(g_hat_y, g_y) for g_hat_y, g_y in zip(vals, int_f_y)]\n",
```